db/migrate/20180103163734_add_index_to_workshops.rb

```diff
@@ -0,0 +1,5 @@
+class AddIndexToWorkshops < ActiveRecord::Migration[5.1]
+  def change
+    add_index(:workshops, [:longitude, :latitude])
```

Comment:
In this case you can't add different kinds to the same place. Add 'type', but i don't thing this is necessary.
Important is adding a single index for 'longitude' and for 'latitude'.